### ISSUE TITLE
Nodegroup failed to create, and now im unable to query nodegroups

### ISSUE THREAD

**Issue by jrich523** [weaveworks/eksctl#4499]
**What were you trying to accomplish?**
Cleanup a failed nodegroup that was unable to create due to a lack of IPs

**What happened?**
the nodegroup ended up in a rolled_back state and running eksctl get nodegroup --cluster <name> 
I unfortnately dont have the orginal error, but after deleting the failed stack I get the following error

Error: getting nodegroup's kubernetes versoin: failed to ist nodes: no nodes were found

**How to reproduce it?**
create a nodegroup that will fail to create, not exactly sure how best to simulate this, since in my case it was a shortage of IPs in the subnet

**Anything else we need to know?**
Ubuntu 18.04

**Versions**
eksctl version: 0.75.0
kubectl version: v1.18.18
OS: linux

Mostly im curious what eksctl is querying to get the list so i know what needs to be cleaned up.

Thanks
Justin

**Comment by cPu1** [comment]
> but after deleting the failed stack I get the following error

>Error: getting nodegroup's kubernetes versoin: failed to ist nodes: no nodes were found

If you're still getting this error, can you share the logs from rerunning the command with verbose logging enabled (passing `-v=4`)?

**Comment by jrich523** [comment]
Its a rather massive log since we have roughly 80 nodegroups.
Essentially it queries the cluster ok, and then it queries the nodegroups, which based on the logging format, completes correctly it seems. Looking at the StackStatus for each, none are listed as in a bad state. I then see a handful of Rate exceeded for cloudformation/GetTemplate and then it spews the error
Error: getting nodegroup's kubernetes versoin: failed to list nodes: no nodes were found

Should i try and check for nodes related to the nodegroup names to see which in the list has no nodes?

Thanks

**Comment by jrich523** [comment]
I think i found the issue, however im not able to figure out why its happened.
I took the log output of that command and pulled out all the nodegroups values. I then compared that to the nodegroup labeles on the nodes, and found that one of the nodegroups didnt have active k8s nodes (or didnt have labels) so i tracked down the ASG and found that the ASG was fine and had 3 nodes, but when look for k8s nodes with a instance label matching the ec2 instance ID i couldnt find that instance. i still need to dig a bit, but i think this is whats causing the error.

I dont really understand how this CF stack could be all good, but these instances not be included as nodes.

**Comment by jrich523** [comment]
Yes that was it. I was able to find the nodegroup that was a valid/healthy stack, that didnt have its instances onboarded as k8s nodes. I was able to do an eksctl delete nodegroup. it detected there was no nodes to drain and didnt care, so that was awesome. 
My only suggestion would be to modify the error message to include the nodegroup that the get was searching for nodes for to help more easily identify what is causing the problem.
Thanks!
